coreneuron/io/reports/report_event.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <algorithm>
+#include <map>
+#include <vector>
+#include "coreneuron/network/netcon.hpp"
+#include "coreneuron/network/netcvode.hpp"
+
+namespace coreneuron {
+
+#if defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+struct VarWithMapping {
+    int id;
+    double* var_value;
+    VarWithMapping(int id_, double* v_) : id(id_), var_value(v_) {
+    }
+};
+
+// mapping the set of variables pointers to report to its gid
+typedef std::map<int, std::vector<VarWithMapping> > VarsToReport;
+
+class ReportEvent : public DiscreteEvent {
+  public:
+    ReportEvent(double dt, double tstart, const VarsToReport& filtered_gids, const char* name);
+
+    /** on deliver, call ReportingLib and setup next event */
+    virtual void deliver(double t, NetCvode* nc, NrnThread* nt);
+    virtual bool require_checkpoint();
```

Comment:
```suggestion
    bool require_checkpoint() override;
```

Besides, `DiscreteEvent::require_checkpoint` should be marked with `const` specifier.